### ISSUE TITLE
Add GAN training

### ISSUE THREAD

**Issue by shahbuland** [OpenWorld-Labs/owl-vaes#1]
Goal: To add GAN training and/or finetuning to the owl vaes

Modelling:
- No real modelling requirements for this task
- A R3GAN discriminator  with hinge loss has been implemented and put in `owl_vaes/discriminators` 
- Note, that it has not been tested in a GAN training setup
- You can see config params in `owl_vaes/discriminators/res.py`

Training:
- When making trainers be sure to add to `owl_vaes/trainers/__init__.py` so you can reference from config
- See configs in `configs` for reference on configuration
- See `owl_vaes/trainers/rec.py` for reference trainer implementation
- Inherit from the base trainer and follow along with reconstruction only trainer where possible 

Full VAEGAN Training: [Optional]  
- Add GAN trainer to `owl_vaes/trainers/gan.py`
- LPIPS + GAN + MSE
- ensure use of delay_adv and warmup_adv from config
- be sure to add trainer to `owl_vaes/trainers/__init__.py` afterwards for easy config access

Decoder Fine Tuning:
- Similar to above but freeze+compile encoder
- Focus purely on training decoder from previous checkpoint (Ask @shahbuland for checkpoints)
- Load checkpoints using config
- LPIPS + GAN + MSE but only for decoder

Notes:
- Discriminator config should be under model config (i.e. as a field inside it)
i.e.
```
model:
    discriminator:
        model_id: resnet
        ch_0: ....
        ....

    model_id: resnet
    ...
```
